lumicks/pylake/population/dwelltime.py

```diff
@@ -669,12 +693,27 @@ def pdf(self, x):
         x : np.array
             array of independent variable values at which to calculate the PDF.
         """
-        return np.exp(
-            _exponential_mixture_log_likelihood_components(
-                self.amplitudes, self.lifetimes, x, *self._observation_limits
-            )
+        limits = np.vstack(list(np.broadcast(*self._observation_limits)))
+        unique_limits, counts = np.unique(limits, axis=0, return_counts=True)
+
+        pdfs = np.asarray(
+            [
+                (count / sum(counts))
+                * np.logical_and(x >= limits[0], x < limits[1])
+                * np.exp(
+                    _exponential_mixture_log_likelihood_components(
+                        self.amplitudes,
+                        self.lifetimes,
+                        x,
+                        *limits,
+                    )
+                )
+                for limits, count in zip(unique_limits, counts)
+            ]
```

Comment:
I think we discussed this briefly, but I'm not sure how I feel about changing the functionality of `.pdf()`. This implementation is specifically for plotting over the histogram, but there are still use cases for evaluating the PDF at a single point rather than over a discrete bin; which by definition at that point you're either below or above any particular observation limit.

Would it maybe make sense to keep the original PDF method, and add this as a specific separate method (probably private)?

Reply:
I'm not sure what you mean by "which by definition at that point you're either below or above any particular observation limit.".

Do you have a concrete use case for it? And which observation limits would you use in that case?

This one isn't specific for histogram. The added complexity is needed for the fact that there can be multiple observation limits.

The old histogram also contained a bug (maybe this needs a changelog entry as well actually 🤔), in the sense that it actually (incorrectly) returned nonzero values beyond the observation limits (meaning that technically, given that you've already taken the observation limits into account in the normalization, it would be incorrectly normalized).

Reply:
yeah I was confusing this with the discretized PDF needed for histogramming (which you addressed in the next comment). It is weird because this PDF is not generalizable (it's set to the fraction of data points with each min/max combo used to train it) ... but there's no real better way to do it, it's just a strange scenario. Realistically based on expected acquisition parameters and biological lifetimes, most people won't even notice this behavior so it's fine to keep it as is.